### ISSUE TITLE
Cannot convert ERC1155TransferBatchEvent to JSValue

### ISSUE THREAD

**Issue by joeykrug** [0xProject/0x-mesh#656]
# Context

Please provide any relevant information about your setup

- Running V3 version in browser dev-utils-fix version

# Expected Behavior

Trading won't crash mesh

# Current Behavior

Trading is crashing mesh.

# Failure Information (for bugs)

https://github.com/AugurProject/augur/pull/5553 is the branch it errors on, it happens when doing a trade. So if you place an order for like 100 shares at 0.50 and then fill it on the book, after the fill mesh throws this error

we never receive the mesh event for the fill
it crashes before then
it's failing to decode this somehow
ERC1155TransferBatchEvent
https://github.com/0xProject/0x-mesh/blob/0xV3/zeroex/orderwatch/decoder/event_decoder_js.go#L70

## Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Download augur repo and checkout the pull request 5553 branch
docker stop $(docker ps -a -q);
docker kill $(docker ps -a -q); 
docker system prune -af
yarn clean
 yarn 
yarn build 
yarn docker:all true 
yarn build -w 
yarn workspace @augurproject/ui dev  
yarn flash run create-reasonable-yes-no-market
yarn flash run faucet -a 100000 -t your_eth_address
2. Do a trade

## Failure Logs

```panic: ValueOf: invalid value
wasm_exec.js:85 
wasm_exec.js:85 goroutine 319 [running]:
wasm_exec.js:85 syscall/js.ValueOf(0xb4740, 0xca5fdc0, 0xca93323)
wasm_exec.js:85     /usr/local/opt/go/libexec/src/syscall/js/js.go:176 +0x141
wasm_exec.js:85 syscall/js.Value.Set(0x7ff8000000003749, 0x200215, 0x3, 0xb4740, 0xca5fdc0)
wasm_exec.js:85     /usr/local/opt/go/libexec/src/syscall/js/js.go:255 +0x2
wasm_exec.js:85 syscall/js.ValueOf(0xfa760, 0xc30f5f0, 0x202d24)
wasm_exec.js:85     /usr/local/opt/go/libexec/src/syscall/js/js.go:172 +0x15
wasm_exec.js:85 github.com/0xProject/0x-mesh/zeroex/orderwatch/decoder.ERC1155TransferBatchEvent.JSValue(0x1b573ccaad2a1515, 0xa4ce0f02e132a30b, 0x70e34967f9a3953f, 0x4f4f92fa5f95b1e7, 0x2c51c75326f1f575, 0xb1d2abc47b47ece4, 0x57bf14ba8cbb2582, 0x2b087f86, 0xfdf048, 0x0, ...)
wasm_exec.js:85     /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/zeroex/orderwatch/decoder/event_decoder_js.go:79 +0x46
wasm_exec.js:85 github.com/0xProject/0x-mesh/zeroex.ContractEvent.JSValue(0xb283764bafe3281b, 0xea1e734ec07abfd8, 0x40308ae1dc5365da, 0xb71733ba70013d2d, 0x64b854172615b149, 0x9da7ffa41fc5d05d, 0x90bbab5c0f853939, 0xf945a7ef2d38d931, 0x0, 0x1, ...)
wasm_exec.js:85
/Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/zeroex/order_js.go:81 +0x7
wasm_exec.js:85 github.com/0xProject/0x-mesh/zeroex.OrderEvent.JSValue(0x0, 0xed5b303f8, 0xfc3e80, 0x29ce1d91d309c296, 0xb932a15675bfd5d4, 0xf20252db58cdaeea, 0x14c55c9384a1897e, 0xccb8a20, 0x207d2e, 0xc, ...)
wasm_exec.js:85     /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/zeroex/order_js.go:16 +0xa
wasm_exec.js:85 main.(*MeshWrapper).Start.func2(0xc07e0f0)
wasm_exec.js:85     /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/browser/go/main.go:197 +0x2a
wasm_exec.js:85 created by main.(*MeshWrapper).Start
wasm_exec.js:85     /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/browser/go/main.go:183 +0x1d```

**Comment by joeykrug** [comment]
browser mesh fails, the mesh rpc client works

in ZeroX.ts in augur you can comment out 

`this.browserMesh = mesh;`

and everything works

**Comment by jalextowle** [comment]
Thank you for the bug report! I am currently working on a PR that adds more robust testing for the browser type conversions, which should address problems like this. I have already squashed a few bugs relating to `OrderEvent.JSValue()`, and these fixes will be available next week.

In the future, it would be helpful if you could share the actual orders so that it's clear what caused `OrderEvent.JSValue()` to fail to decode and we can add them as test cases.

**Comment by albrow** [comment]
This should be fixed in #659, which was included in version 8.1.2.
